src/Datadog.Trace/FrameworkDescription.cs

```diff
@@ -0,0 +1,156 @@
+            string processArchitecture)
+        {
+            Name = name;
+            ProductVersion = productVersion;
+            FileVersion = fileVersion;
+            AssemblyVersion = assemblyVersion;
+            EnvironmentVersion = environmentVersion;
+            OSPlatform = osPlatform;
+            OSArchitecture = osArchitecture;
+            ProcessArchitecture = processArchitecture;
+        }
+
+        public static FrameworkDescription Instance
+        {
+            get
+            {
+                if (_instance == null)
+                {
+                    _instance = Create();
+                }
+
+                return _instance;
+            }
+        }
+
+        public string Name { get; }
+
+        public string ProductVersion { get; }
+
+        public string FileVersion { get; }
+
+        public Version AssemblyVersion { get; }
+
+        public Version EnvironmentVersion { get; }
+
+        public string OSPlatform { get; }
+
+        public string OSArchitecture { get; }
+
+        public string ProcessArchitecture { get; }
+
+        public static string GetOSPlatform()
+        {
+            switch (Environment.OSVersion.Platform)
+            {
+                case PlatformID.Win32NT:
+                    return "win";
+                case PlatformID.Unix:
+                    return "linux";
+                default:
+                    return "unknown";
+            }
+        }
+
+        public override string ToString()
+        {
+            string architecture;
+
+            if (OSPlatform == "win" && ProcessArchitecture == "x86" && OSArchitecture == "x64")
+            {
+                architecture = "win-wow64";
+            }
+            else
+            {
+                architecture = $"{OSPlatform}-{OSArchitecture}";
+            }
+
+            return $"{Name} {ProductVersion} {architecture}";
+        }
+
+        private static FrameworkDescription Create()
+        {
+            var assembly = typeof(object).GetTypeInfo().Assembly;
+            var assemblyName = assembly.GetName();
+
+            var fileVersionAttribute = (AssemblyFileVersionAttribute)assembly.GetCustomAttribute(typeof(AssemblyFileVersionAttribute));
+            var fileVersion = fileVersionAttribute.Version;
+
+            var osArchitecture = Environment.Is64BitOperatingSystem
+                                     ? "x64"
+                                     : "x86";
+            var processArchitecture = Environment.Is64BitProcess
+                                          ? "x64"
+                                          : "x86";
+
+            if (string.Equals(assemblyName.Name, "mscorlib", StringComparison.OrdinalIgnoreCase))
+            {
+                return new FrameworkDescription(
+                    ".NET Framework",
+                    fileVersion,
+                    fileVersion,
+                    assemblyName.Version,
+                    Environment.Version,
```

Comment:
For .NET Framework, we might want to use the registry instead of querying the `Environment.Version` property, as noted here: https://docs.microsoft.com/en-us/dotnet/api/system.environment.version?view=netframework-4.8#remarks

Reply:
Yeah, this was leftover from very WIP code where I was capturing all the possible versions (file version, assembly version, `Environment.Version`, etc) for comparison. We weren't using several of them so I removed them. Thanks.